### PR TITLE
QASM->Q# translator into v0.3

### PR DESCRIPTION
**Pull request by anpaz-msft**
Q# 0.3 introduced some breaking syntax changes and changes to the repo's folder structure; unfortunately we didn't have time to merge all external contributions from master into this release. We're now finally working on getting these contributions migrated, once all are migrated we'll merge `release/v0.3.1810` back into `master` and make `master` again the default branch.
These merges #58 into the release branch. No new functionality is introduced, just syntax and folder changes.

### CODE REVIEW

**Comment by anpaz-msft** [comment]
@RolfHuisman, could you please take a look and make sure I'm missing nothing.
Thanks!

File: Samples/src/OpenQasmReader/Parser.cs [hunk 1]
```diff
@@ -305,7 +305,7 @@ private static void ParseGateCall(IEnumerator<string> token, StringBuilder build
                 }
                 else if (withinParentheses)
                 {
-                    doubles.Add(ParseCalculation(token, COMMA, CLOSE_PARANTHESES));
+                    doubles.Add(ParseCalculation(token, $"{COMMA} ", CLOSE_PARANTHESES));
```
Comment (RolfHuisman) [inline Samples/src/OpenQasmReader/Parser.cs]: Would it make sense to make this a constant of its own?

**Comment by RolfHuisman** [comment]
Was missing the changes of #88, but seems that one hasn't been merged yet.

File: Samples/src/OpenQasmReader/Parser.cs (hunk 1 above)
Comment (anpaz-msft) [reply Samples/src/OpenQasmReader/Parser.cs]: I'll let you do that later ;p

**Pull request merged**